File: Python/Data Structures & Algorithms Module - Python/Week7/Factorial.py
"""
Factorial.py
Billy Cussen
17/11/2020
"""
def factorial(num):
    res = num
    while num != 1:
        num-=1
        res*=num
    return res

def factorialRecursion(num):
    if num == 1 or num == 0:
        return num
    return num * factorialRecursion(num-1)

print("Factorial: "+str(factorial(5)))
print("Factorial Recursion: "+str(factorialRecursion(5)))

File: Java/Data Structures & Algorithms Module - Java/Week8/Revision/Week7/Factorial.java
/*
Factorial.java
Billy Cussen
17/11/2020
*/

public class Factorial{

    public static void main(String[] args){

        System.out.println(factorial(5));
        System.out.println(factorialRecursion(5));
    }

    public static int factorial(int n){
        int res = n;
        while(n>1){
            n-=1;
            res*=n;
        }
        return res;
    }

    public static int factorialRecursion(int n){
        if (n == 1){
            return n;
        }
        return n * factorialRecursion(n-1);
    }
}